src/test/java/org/opensearch/neuralsearch/query/HybridQueryPostFilterIT.java

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.SneakyThrows;
+import org.junit.Before;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.MatchNoneQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregations;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getNestedHits;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregationValues;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.aggregations.AggregationBuilders;
+import static org.opensearch.neuralsearch.util.TestUtils.assertHitResultsFromQuery;
+
+public class HybridQueryPostFilterIT extends BaseNeuralSearchIT {
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_MULTIPLE_SHARDS =
+        "test-hybrid-post-filter-multi-doc-index-multiple-shards";
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_SINGLE_SHARD =
+        "test-hybrid-post-filter-multi-doc-index-single-shard";
+    private static final String SEARCH_PIPELINE = "phase-results-hybrid-post-filter-pipeline";
+    private static final String INTEGER_FIELD_1 = "stock";
+    private static final String TEXT_FIELD_1 = "name";
+    private static final String KEYWORD_FIELD_2 = "category";
+    private static final String TEXT_FIELD_NAME_1_VALUE = "Dunes part 2";
+    private static final String TEXT_FIELD_NAME_2_VALUE = "Dunes part 1";
```

Comment:
```suggestion
    private static final String TEXT_FIELD_NAME_2_VALUE = "Dunes part 1";
```
can this be changed to List or Map instead of individual variables?

Reply:
I think Using single variable is much easier because we use these variables in many places then if use any collection then we have to do lot of get(index) calls from that collection and code with look more clumpsy.